src/bitcoind.ts

```diff
@@ -87,10 +92,22 @@ export class BitcoindClient {
     hexstring,
   }: {
     hexstring: string
-  }): Promise<{ vout: [VOut] }> {
+  }): Promise<DecodeRawTransactionResult> {
     return await this.client.decodeRawTransaction({ hexstring })
   }
 
+  // default client knows about all transactions of its wallets
+
+  async getTransaction({
+    txid,
+    include_watchonly,
+  }: {
+    txid: string
+    include_watchonly?: boolean
+  }): Promise<InWalletTransaction> {
+    return await this.client.getTransaction({ txid, include_watchonly })
```

Comment:
Why are you sometimes using the default client and sometimes `this.client`? I think I'd prefer to always use `this.client` and pass in the `default` client via constructor if that works.

Reply:
`this.client` is also the instance variable name used in the BitcoindWalletClient because it actually is a client just like in the BitcoindClient class. The only difference is that in BitcoindWalletClient it was instantiated with the `wallet` argument. This is the way for the BitcoindWalletClient instances to contact the same "default" (main) client while being "loaded" and ready to invoke wallet rpc calls.